tests/gensim/test_gensim_model_export.py

```diff
@@ -0,0 +1,236 @@
+    assert os.path.exists(saved_conda_env_path)
+
+    with open(saved_conda_env_path, "r") as f:
+        saved_conda_env_parsed = yaml.safe_load(f)
+    assert saved_conda_env_parsed == conda_env
+
+
+@pytest.mark.large
+def test_model_log_persists_specified_conda_env_in_mlflow_model_directory(
+        gensim_word2vec_model, gensim_custom_env):
+    artifact_path = "model"
+    with mlflow.start_run():
+        mlflow.gensim.log_model(gen_model=gensim_word2vec_model.model,
+                                artifact_path=artifact_path,
+                                conda_env=gensim_custom_env)
+        model_uri = "runs:/{run_id}/{artifact_path}".format(
+            run_id=mlflow.active_run().info.run_id,
+            artifact_path=artifact_path)
+
+    model_path = _download_artifact_from_uri(artifact_uri=model_uri)
+    pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
+    saved_conda_env_path = os.path.join(model_path, pyfunc_conf[pyfunc.ENV])
+    assert os.path.exists(saved_conda_env_path)
+    assert saved_conda_env_path != gensim_custom_env
+
+    with open(gensim_custom_env, "r") as f:
+        gensim_custom_env_parsed = yaml.safe_load(f)
+    with open(saved_conda_env_path, "r") as f:
+        saved_conda_env_parsed = yaml.safe_load(f)
+    assert saved_conda_env_parsed == gensim_custom_env_parsed
+
+
+@pytest.mark.large
+def test_model_save_without_specified_conda_env_uses_default_env_with_expected_dependencies(
+        gensim_word2vec_model, model_path):
+    word2vec_model = gensim_word2vec_model.model
+    mlflow.gensim.save_model(gen_model=word2vec_model, path=model_path, conda_env=None)
+
+    pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
+    conda_env_path = os.path.join(model_path, pyfunc_conf[pyfunc.ENV])
+    with open(conda_env_path, "r") as f:
+        conda_env = yaml.safe_load(f)
+
+    assert conda_env == mlflow.gensim.get_default_conda_env()
+
+
+@pytest.mark.large
+def test_model_log_without_specified_conda_env_uses_default_env_with_expected_dependencies(
+        gensim_word2vec_model):
+    artifact_path = "model"
+    word2vec_model = gensim_word2vec_model.model
+    with mlflow.start_run():
+        mlflow.gensim.log_model(gen_model=word2vec_model,
+                                artifact_path=artifact_path,
+                                conda_env=None)
+
+        model_uri = "runs:/{run_id}/{artifact_path}".format(
+            run_id=mlflow.active_run().info.run_id,
+            artifact_path=artifact_path)
+
+    model_path = _download_artifact_from_uri(artifact_uri=model_uri)
+    pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
+    conda_env_path = os.path.join(model_path, pyfunc_conf[pyfunc.ENV])
+    with open(conda_env_path, "r") as f:
+        conda_env = yaml.safe_load(f)
+
+    assert conda_env == mlflow.gensim.get_default_conda_env()
```

Comment:
Let's make sure we test `pyfunc.load_model` as well (see `test_tensorflow_model_export.py` for an example). We want to make sure this flavor follows the same API as other pyfunc flavors.